client/my-sites/plugins/plugin-details.jsx

```diff
@@ -436,6 +441,15 @@ function onClickInstallPlugin( { dispatch, selectedSite, slug, upgradeAndInstall
 		} )
 	);
 
+	dispatch( productToBeInstalled( null, slug, selectedSite.slug ) );
+
+	dispatch(
+		pluginInstallationStateChange(
+			MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS,
+			'preauthorize plugin installation URL'
+		)
+	);
+
```

Comment:
We will need to add similar events to the Plugin Upload flow `https://wordpress.com/plugins/upload/[DOMAIN]` which is now returning an error. 

https://github.com/Automattic/wp-calypso/blob/08961f239b9c353bb0b44983d9f7b014db038133/client/my-sites/plugins/plugin-upload/index.jsx#L60-L62

Reply:
Thanks @cpapazoglou, that's a very good point!